test/tests.jl

```diff
@@ -212,6 +223,146 @@ end
+            str = String(take!(buf))
+            @test occursin("\n```julia\n", str)
+            @test occursin("k_2(x::String, y::U<:Complex, z::T<:Number) -> String", str)
+            @test occursin("\n```\n", str)
+
+            doc.data = Dict(
+                :binding => Docs.Binding(M, :k_3),
+                :typesig => (Union{Tuple{Any, T, U}, Tuple{U}, Tuple{T}} where U <: Any) where T <: Any,
+                :module => M,
+            )
+            DSE.format(DSE.TYPEDSIGNATURES, buf, doc)
+            str = String(take!(buf))
+            @test occursin("\n```julia\n", str)
+            @test occursin("\nk_3(x::Any, y::T, z::U) -> Any\n", str)
+            @test occursin("\n```\n", str)
+
+            doc.data = Dict(
+                :binding => Docs.Binding(M, :k_4),
+                :typesig => Union{Tuple{String}, Tuple{String, Int}},
+                :module => M,
+            )
+            DSE.format(DSE.TYPEDSIGNATURES, buf, doc)
+            str = String(take!(buf))
+            @test occursin("\n```julia\n", str)
+            if VERSION > v"1.3.0"
+                @test occursin("\nk_4(::String)\n", str)
+                if typeof(1) === Int64
+                    @test occursin("\nk_4(::String, ::Int64)\n", str)
+                else
+                    @test occursin("\nk_4(::String, ::Int32)\n", str)
+                end
+            else
+                # TODO: remove this test when julia 1.0.0 support is dropped.
+                # older versions of julia seem to return this
+                # str = "\n```julia\nk_4(#temp#::String)\nk_4(#temp#::String, #temp#::Int64)\n\n```\n\n"
+                @test occursin("\nk_4", str)
+            end
+            @test occursin("\n```\n", str)
+
+
+            doc.data = Dict(
+                :binding => Docs.Binding(M, :k_5),
+                :typesig => Union{Tuple{Type{T}, String}, Tuple{Type{T}, String, Union{Nothing, Function}}, Tuple{T}} where T <: Number,
+                :module => M,
+            )
+            DSE.format(DSE.TYPEDSIGNATURES, buf, doc)
+            str = String(take!(buf))
+            @test occursin("\n```julia\n", str)
+            if VERSION > v"1.3.0"
+                @test occursin("\nk_5(::Type{T<:Number}, x::String) -> String\n", str)
+                @test occursin("\nk_5(::Type{T<:Number}, x::String, func::Union{Nothing, Function}) -> String\n", str)
+            end
+            @test occursin("\n```\n", str)
+
+            doc.data = Dict(
+                :binding => Docs.Binding(M, :k_6),
+                :typesig => Union{Tuple{Vector{T}}, Tuple{T}} where T <: Number,
+                :module => M,
+            )
+            DSE.format(DSE.TYPEDSIGNATURES, buf, doc)
+            str = String(take!(buf))
+            @test occursin("\n```julia\n", str)
+            if VERSION > v"1.3.0"
+                @test occursin("\nk_6(x::Array{T<:Number,1}) -> Array{T<:Number,1}\n", str)
+            end
```

Comment:
Why are we version-guarding these tests? Ditto for the tests below.

Reply:
I've added a version guard for the only function that requires it in the test. Resolved in 403028e.